core/main/src/firebolt/firebolt_gateway.rs

```diff
@@ -278,11 +336,18 @@ impl FireboltGateway {
                         }
                     }
 
-                    if !platform_state.endpoint_state.handle_brokerage(
+                    let requestor_callback_tx =
+                        Self::handle_broker_callback(platform_state.clone(), request_c.clone());
```

Comment:
Nit: You may pass these parameters by reference to avoid cloning.

Reply:
Need to clone as they are moved to another thread.